src/sema/statements.rs

```diff
@@ -888,17 +888,38 @@ fn statement(
                 return Err(());
             }
 
+            let mut memory_safe = None;
+
             if let Some(flags) = flags {
                 for flag in flags {
-                    ns.diagnostics.push(Diagnostic::error(
-                        flag.loc,
-                        format!("flag '{}' not supported", flag.string),
-                    ));
+                    if flag.string == "memory-safe" && ns.target == Target::EVM {
```

Comment:
Why is there no enum and the comparison is on `flag.string`?

Reply:
assembly statements can have flags, like so:
```solidity
function foo() { 
     assembly ("foo", "bar", "baz") {
         // ..
     }
}
```
according to the solidity parser/grammer. Our parser just parses it into a list of flags (string literals).

However there is only one flag defined which is `"memory-safe"`. The change is to parse this for EVM and add it to the AST, we don't allow the flag for any other target and we don't use it on EVM either - as our EVM target is incomplete.

Reply:
> Our parser just parses it into a list of flags (string literals).

So in sema, we can parse this list of string literals parsed into a `HashSet` or`BTreeSet` of an `AssemblyFlag` enum right? This will make using it in later compile stages in the future easier.

Reply:
Right now there is only one valid flag: `memory-safe`. I don't think it's worth creating a `HashSet` for that yet. We can implement that should more flags be added.